### ISSUE TITLE
Mutators without flyout blocks breaks with NPE on render

### ISSUE THREAD

**Issue by mattbishop** [google/blockly#3955]
**Describe the bug**

I am creating a mutator with a single root block containing fields. I do not have any statement inputs or connections to add from the flyout.

In my mutable block's input() method, I call `this.setMutator(new Blockly.Mutator([]));` to set up the mutator icon and logic. This block contains the requisite mutator mixing methods.

When I click the mutator, I see my root block render as expected, but the bubble is huge and never resizes to match. I get an NPE in the console:

`TypeError: null is not an object (evaluating 'flyoutMetrics.contentHeight')`
in mutators.js:231

I debugged this method and found that flyoutMetrics is null because the flyout is not visible so flyout.getMetrics() returns null.

The bug, I think, is that the flyout is never init()ed if the workspace's languageTree is null, which it is because the quarks list is empty. From mutator.js:286

```
    var tree = this.workspace_.options.languageTree;
    var flyout = this.workspace_.getFlyout();
    if (tree) {
      flyout.init(this.workspace_);
      flyout.show(tree.childNodes);
    }
````

**Expected behavior**

I should be able to create a mutator with a single root block containing fields and not coming up with a way to use statement and connectors to express mutation configuration.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: MacOS 10.15.5
 - Browser chrome, safari
 - Version chrome 83

**Stack Traces**

```
[Error] TypeError: null is not an object (evaluating 'flyoutMetrics.contentHeight')
	(anonymous function) (mutator.js:231)
	(anonymous function) (mutator.js:323)
	(anonymous function) (icon.js:124)
	(anonymous function) (mutator.js:124)
	wrapFunc (blockly.js:469)
```

**Comment by samelhusseini** [comment]
Just to confirm I understand what you're trying to achieve: 
you want to show the mutator UI with no flyout, and just a single root block (that you can then perhaps use fields) to configure your mutation.

**Comment by mattbishop** [comment]
Yes. I monkey-patched Blockly.Mutator.prototype.resizeBubble_ to test for a null flyoutMetrics and it renders as I hoped. I changed line 230 to this:

```
  if (flyout && flyout.getMetrics_()) {. // <-- flyout is not null, but getMetrics() returns null
    var flyoutMetrics = flyout.getMetrics_();
    height = Math.max(height, flyoutMetrics.contentHeight + 20);
  }
```

**Comment by samelhusseini** [comment]
Fixed by #3958
